core/store/src/db/rocksdb.rs

```diff
@@ -466,7 +466,19 @@ fn rocksdb_read_options() -> ReadOptions {
     read_options
 }
 
-fn rocksdb_block_based_options(
+fn rocksdb_block_based_options_no_index_cache(
```

Comment:
So the reason I would like to have two separate functions for different Block TableOptions is that they represent very different configuration settings for column families, since `rocksdb_block_based_options_no_index_cache` will be used for both `FlatState` and `State` and `rocksdb_block_based_options_default` for rest of the columns. Even though the code is similar and could be extracted into a single function for the code's sake, my opinion is not to do that since we would want to keep these configurations separately.

Reply:
+1 to Yoon's comment, I think it would be better to keep it as a single function and introduce a new argument `set_cache_index_and_filter_blocks: bool` argument.